scanner/build.gradle.kts

```diff
@@ -85,3 +90,11 @@ buildConfig {
     buildConfigField("String", "SCANCODE_VERSION", "\"$scancodeVersion\"")
     buildConfigField("String", "SCANOSS_VERSION", "\"$scanossVersion\"")
 }
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+    if (name.contains("Test")) {
```

Comment:
I believe it's fine to unconditionally do this for all tasks. That would also align with existing similar code.

Reply:
Speaking of aligning code, please also use the `customCompilerArgs` approach here.

Reply:
> I believe it's fine to unconditionally do this for all tasks. That would also align with existing similar code.

It does not work if I remove the conditional: the `kotlinx` dependency being a test dependency (no release runtime code uses it), it tries to build non-test classes with the `optIn` and fails.

Reply:
Interesting. Then please at lest use `"test" in name.lowerCase()` instead.

Reply:
Done but Intellij does not recognize `lowerCase`: I am forced to use the Java version. Any idea ?

Reply:
Ah, might be that for Gradle KTS files we're stuck with a Kotlin version where it's still called `toLowerCase()`.

Reply:
Can we merge this ?